Failed to compile on Windows 10

Hi,

I'm trying to test the project on Windows 10 (20H2) with rustc 1.56.1 and it failed at:

```
error[E0308]: mismatched types
  --> src/bin/elfshaker\store.rs:83:8
   |
83 |     Ok(std::str::from_utf8(buf))
   |        ^^^^^^^^^^^^^^^^^^^^^^^^ expected `&OsStr`, found enum `Result`
   |
   = note: expected reference `&OsStr`
                   found enum `Result<&str, std::str::Utf8Error>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `elfshaker` due to previous error
```

I tried and fixed the issue by creating a new `OsStr` and it seems that it worked. Everything compiled and running.

```
#[cfg(not(unix))]
fn to_os_str(buf: &[u8]) -> Result<&OsStr, std::str::Utf8Error> {
    // On Windows (and everything else) we will expect well-formed UTF-8 and pray
    Ok(OsStr::new(std::str::from_utf8(buf)?))
}
```

However, I don't know whether that is a correct solution for this. The executable runs on my machine and two out of the 18 test failed. I saw in the README that it's only tested with Ubuntu so I guess the `building_loose_object_paths_works` test failed because of that. I don't know why `test_atomic_update_api` failed, however.

```
running 18 tests
test repo::algo::test::partition_by_u64_small_values_works ... ok
test repo::algo::test::partition_by_u64_few_works ... ok
test repo::algo::test::partition_by_u64_simple_works ... ok
test repo::pack::tests::assign_to_frames_multiple_works ... ok
test entrypool::tests::basic_pathpool_test ... ok
test entrypool::tests::serde ... ok
test repo::pack::tests::assign_to_frames_single_works ... ok
test repo::pack::tests::pack_id_validation_works ... ok
test repo::pack::tests::snapshot_tag_validation_works ... ok
test repo::repository::tests::building_loose_object_paths_works ... FAILED
test repo::repository::tests::compute_entry_diff_finds_update_of_duplicated ... ok
test repo::repository::tests::compute_entry_diff_finds_updates ... ok
test repo::repository::tests::compute_entry_diff_path_switch ... ok
test repo::repository::tests::data_dir_detected ... ok
test repo::repository::tests::data_dir_detected_as_parent ... ok
test repo::repository::tests::data_dir_not_detected_incorrectly ... ok
test repo::algo::test::run_in_parallel_works ... ok
test atomicfile::tests::test_atomic_update_api ... FAILED

failures:

---- repo::repository::tests::building_loose_object_paths_works stdout ----
thread 'repo::repository::tests::building_loose_object_paths_works' panicked at 'assertion failed: `(left == right)`
  left: `"/repo/elfshaker_data/loose/fa/f0/deadbeefbadc0de0faf0deadbeefbadc0de0"`,
 right: `"/repo\\elfshaker_data\\loose\\fa\\f0\\deadbeefbadc0de0faf0deadbeefbadc0de0"`', src\repo\repository.rs:841:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- atomicfile::tests::test_atomic_update_api stdout ----
thread 'atomicfile::tests::test_atomic_update_api' panicked at 'unexpected error: Os { code: 5, kind: PermissionDenied, message: "Access is denied." }', src\atomicfile.rs:206:23


failures:
    atomicfile::tests::test_atomic_update_api
    repo::repository::tests::building_loose_object_paths_works

test result: FAILED. 16 passed; 2 failed; 0 ignored; 0 measured; 0 filtered out; finished in 2.93s
```

Thank you so much for your time. This is a really great project.

--- a/src/bin/elfshaker/store.rs
+++ b/src/bin/elfshaker/store.rs
@@ -80,7 +80,7 @@ fn to_os_str(buf: &[u8]) -> Result<&OsStr, std::str::Utf8Error> {
 #[cfg(not(unix))]
 fn to_os_str(buf: &[u8]) -> Result<&OsStr, std::str::Utf8Error> {
     // On Windows (and everything else) we will expect well-formed UTF-8 and pray
-    Ok(std::str::from_utf8(buf))
+    Ok(OsStr::new(std::str::from_utf8(buf)?))
 }
 
 fn read_files_list(mut reader: impl io::Read, separator: u8) -> io::Result<Vec<PathBuf>> {

